src/modules/70-pipeline/components/PipelineSteps/Steps/Barrier/Barrier.tsx

```diff
@@ -233,18 +237,59 @@ function BarrierInputStep({ inputSetData, allowableTypes, stepViewType }: Barrie
       parentEntityRepoName: repoName
     }
   })
+
+  const {
+    data: pipelineTemplateResponse,
+    refetch: refetchPipelineTemplate,
+    loading: pipelineTemplateLoading
+  } = useGetTemplate({
+    templateIdentifier: pipeline?.pipeline?.template?.templateRef || '',
+    queryParams: {
+      accountIdentifier: accountId,
+      orgIdentifier,
+      projectIdentifier,
+      repoIdentifier,
+      branch,
+      parentEntityConnectorRef: connectorRef,
+      parentEntityRepoName: repoName
+    },
+    lazy: true
+  })
+
   React.useEffect(() => {
     if (pipelineResponse?.data?.yamlPipeline) {
       setPipeline(parse(pipelineResponse?.data?.yamlPipeline))
     }
   }, [pipelineResponse?.data?.yamlPipeline])
+
+  React.useEffect(() => {
+    if (pipeline?.pipeline?.template?.templateRef) {
+      refetchPipelineTemplate()
+    }
+  }, [pipeline?.pipeline?.template?.templateRef])
+
+  React.useEffect(() => {
+    if (pipelineTemplateResponse?.data && !pipelineTemplateLoading) {
+      const templateSpec = parse<{ template: { spec: StepElementConfig } }>(
+        defaultTo(pipelineTemplateResponse?.data?.yaml, '')
+      )
+      setPipelineTemplate(templateSpec as any)
+    }
+  }, [pipelineTemplateResponse?.data, pipelineTemplateLoading])
```

Comment:
Can we remove pipelineTemplate and useEffect and use memoized value 
```suggestion
const pipelineTemplate = React.useMemo(() => {
  if (pipelineTemplateResponse?.data && !pipelineTemplateLoading) {
    const templateSpec = parse<{ template: { spec: StepElementConfig } }>(
      defaultTo(pipelineTemplateResponse?.data?.yaml, '')
    )
    return templateSpec as any;
  }
  return undefined;
}, [pipelineTemplateResponse?.data, pipelineTemplateLoading]);
```